dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -19,70 +20,82 @@ beforeEach(() => {
   };
 });
 
-it("renders a loading message if apps object is empty", () => {
-  const wrapper = shallow(<AppList {...defaultProps} />);
-  expect(wrapper.text()).toBe("Loading");
-});
+describe("while fetching apps", () => {
+  beforeEach(() => {
+    props = { ...defaultProps, apps: { isFetching: true } };
+  });
 
-it("renders a loading message if it's fetching apps", () => {
-  const wrapper = shallow(
-    <AppList
-      {...defaultProps}
-      apps={
-        {
-          isFetching: true,
-          items: [],
-          listOverview: [],
-          listingAll: false,
-        } as IAppState
-      }
-    />,
-  );
-  expect(wrapper.text()).toContain("Loading");
+  it("matches the snapshot", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("renders a loading message", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.text()).toContain("Loading");
+  });
+
+  it("renders a Application header", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.find("h1").text()).toContain("Applications");
+  });
+
+  it("does not show the search filter nor deploy button", () => {
```

Comment:
do we want to ensure this? Maybe we should show those elements even if listOverview is not yet set

Reply:
This represents the first time that you load the page, not when we search or change the visibility toggle, in which case the controls are actually shown.

I believe that I did this so if the first time it loads the API fails we do not get the controls. Maybe it might make more sense to not to show them if error instead.